### ISSUE TITLE
Switch from using the deprecated `SourceAsset` to `AssetSpec`

### ISSUE THREAD

**Issue by bendnorman** [catalyst-cooperative/pudl#3977]
Switch from using the deprecated [`SourceAsset`](https://docs.dagster.io/_apidocs/assets#dagster.SourceAsset) to [`AssetSpec`](https://docs.dagster.io/concepts/assets/external-assets). 

The FERC Form 1 and Form 714 raw data are extracted into SQLite files before the main ETL begins. To bring the raw SQLite tables into dagster, we create `SourceAsset`s for all the tables. To create a `SourceAsset` you just need to initialize it with a table name and an io manager key.

Dagster is removing `SourceAsset`s in 2.0.0 so we need to start using `AssetSpec`s! This is what the `SourceAsset` docs say:

> Use AssetSpec instead. If using the SourceAsset io_manager_key property, use AssetSpec(...).with_io_manager_key(...)..

Here is an example of how to make this change:
https://github.com/catalyst-cooperative/pudl/blob/9dd1ab9fe483c58541547703b6bd85c922a2aaa5/src/pudl/extract/ferc1.py#L360-L366

to 

```python
raw_ferc1_dbf_assets = [
    AssetSpec(
        key=AssetKey(f"raw_ferc1_dbf__{table_name}"),
    ).with_io_manager_key(""ferc1_dbf_sqlite_io_manager")
    for table_name in dbf_table_names
]
```

```[tasklist]
### Tasks
- [ ] Replace SourceAsset with AssetSpec for FERC 714 and extraction
- [ ] Incorporate AssetSpec into `get_asset_keys()`
```

**Comment by zaneselvans** [comment]
I made a brief attempt at this a couple of weeks ago and ran into some issues. It didn't seem quite as straightforward as they made it sound. See the `deprecate-source-assets` branch for where I left it.

**Comment by bendnorman** [comment]
@ggurjar333 are you still interested in taking this on?

**Comment by ggurjar333** [comment]
Yes I have created a branch on this I'll push it today. But I'm still facing some issues on AssetSpec configuration

**Comment by ggurjar333** [comment]
https://github.com/catalyst-cooperative/pudl/pull/3983
